distutils/cygwinccompiler.py

```diff
@@ -57,6 +57,8 @@ def get_msvcr():
             return ['ucrt', 'vcruntime140']
         else:
             raise ValueError("Unknown MS Compiler version %s " % msc_ver)
+    else:
+        return []
```

Comment:
Here you change the return type of this method and the test that protects the return type, but it's unclear to me why without digging into it a bit. It would be nice if there was a test that would fail if self.dll_libraries is None.